extensions/mssql/src/main.ts

```diff
@@ -134,6 +134,55 @@ export async function activate(context: vscode.ExtensionContext): Promise<MssqlE
 	context.subscriptions.push(new DeleteFilesCommand(prompter, appContext));
 	context.subscriptions.push({ dispose: () => languageClient.stop() });
 
+	azdata.ui.registerModelViewProvider('bdc-endpoints', async (view) => {
+
+		const endpointsArray: Array<Utils.IEndpoint> = view.serverInfo.options['clusterEndpoints'];
+		const managementProxyEp = view.serverInfo.options['clusterEndpoints'].find(e => e.serviceName === 'management-proxy');
```

Comment:
This could be null - you've already got the `endpointsArray` which is the same - please use this, and check it for null & don't do anything unless it exists.

Reply:
or are you trying to clone the clusterEndpoints and push new data to it?
- I'd suggest doing an `endpointsArray = Object.assign({}, endpointsArray)` if you're trying to do this.